lib/wiki_assignment_output.rb

```diff
@@ -0,0 +1,97 @@
+require './lib/wiki'
+#= Class for generating wikitext for updating assignment details on talk pages
+class WikiAssignmentOutput
+  ################
+  # Entry points #
+  ################
+  def self.build_talk_page_update(title,
+                                  talk_title,
+                                  title_assignments,
+                                  course_page)
+    initial_page_content = Wiki.get_page_content talk_title
+    # We only want to add assignment tags to non-existant talk pages if the
+    # article page actually exists. This also servces to make sure that we
+    # only post to talk pages of mainspace articles, as we assume that pages
+    # like Talk:Template:Citation or Talk:Wikipedia:Notability do not exist.
+    if initial_page_content.nil?
+      return nil if Wiki.get_page_content(title).nil?
+      initial_page_content = ''
+    end
+
+    # Limit it to live assignments for this article-course
+    title_assignments = title_assignments.select { |a| !a['deleted'] }
```

Comment:
This would be better expressed as

`title_assignments = title_assignments.reject { |a| a['deleted'] }`

to avoid the negation. It's easier to think of you rejecting the ones that are deleted than selecting the ones that aren't deleted.